src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/SlingInitialContentBundleEntryMetaDataCollector.java

```diff
@@ -190,6 +260,21 @@ private SlingInitialContentBundleEntryMetaData createSlingInitialContentBundleEn
         String repositoryPath = (target != null ? target : "/") + URLDecoder.decode(entryName.substring(pathEntryValue.getPath().length()), "UTF-8");
         return new SlingInitialContentBundleEntryMetaData(targetFile, pathEntryValue, repositoryPath);
     }
+    
+
+    private PathEntry getPathEntryFromRepositoryPath(@NotNull String repositoryPath) {
+
+        if(StringUtils.countMatches(repositoryPath, '/') < 2){
+            //we don't add the primary type of level 1 and 2.
+            return null;
+        }
+        
+        Optional<PathEntry> pathEntryOptional = context.getPathEntryList().stream().filter(
+                pathEntry -> checkIfPathStartsWithOrIsEqual(pathEntry.getTarget(), repositoryPath)
```

Comment:
instead of creating `checkIfPathStartsWithOrIsEqual` yourself, i would strongly recommend to use again Text utility. `isDescendantOrEqual`